ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/reloperators/HiveValues.java

```diff
@@ -43,4 +45,18 @@ public HiveValues(RelOptCluster cluster, RelDataType rowType, ImmutableList<Immu
   public RelNode copy(RelTraitSet traitSet, List<RelNode> inputs) {
     return new HiveValues(getCluster(), getRowType(), tuples, getTraitSet());
   }
+
+  public RelNode copy(List<String> newColumnNames) throws CalciteSemanticException {
+    if (newColumnNames.size() != getRowType().getFieldCount()) {
+      throw new CalciteSemanticException("The number of new column names and columns in the schema does not match!");
+    }
+
+    RelDataTypeFactory.Builder builder = getCluster().getTypeFactory().builder();
```

Comment:
Maybe you can exploit the `builder.uniquify()` method to make the code slightly more readable.

Reply:
Refactored this to use `RelDataTypeFactory.Builder.uniquify()`. Thanks